### PR TITLE
Solve #17 "trading and cancelling" bug

### PR DESCRIPTION
**Pull request by JimZhouZZY**
Solve #17 "Send gold -> Cancel -> Move" bug.
Thank to @beibarys98's commitment, the bug when trading was fixed, but the bug when sending gold still exists. (See #17)
Now it is fixed by a method similar to @beibarys98's.
By the way, corrected and added comments for new variable names.

### CODE REVIEW

File: src/scenes/level_scene.py [hunk 1]
```diff
@@ -2126,6 +2131,15 @@ def right_click(self) -> None:
...
                             self.traded_items.clear()
+                        if self.traded_gold:
+                            # Return traded gold
+                            for gold in self.traded_gold:
+                                if gold[1] == self.selected_player:
+                                    self.selected_player.gold += gold[0]
+                                    gold[2].gold -= gold[0]
+                                else:
+                                    self.selected_player.gold -= gold[0]
+                                    gold[2].gold += gold[0]
```
Comment (Grimmys) [inline src/scenes/level_scene.py:2142]: You are missing the line to clear the `traded_gold` collection.

As a consequence, it's possible to cancel the turn of the same character again and again and the gold is piling up for one character, and the other one starts to have big debts 🥲

**Review by Grimmys — CHANGES_REQUESTED** [review]

File: src/scenes/level_scene.py (hunk 1 above)
Comment (JimZhouZZY) [reply src/scenes/level_scene.py:2142]: Thank you for your careful review, I am very sorry for making this low-level mistake.

Comment (Grimmys) [reply src/scenes/level_scene.py:2142]: I'm also making this kind of mistakes on daily basis, don't worry about it.

**Review by Grimmys — APPROVED** [review]
I'm not able to reproduce the issue anymore, LGTM! :rocket:

**Pull request merged**